### PR TITLE
Implement Transproc::Transformer.define method

### PR DESCRIPTION
**Pull request by Kukunin**

### CODE REVIEW

**Comment by Kukunin** [comment]
I've added inheritance of current transproc on transformer inheritance. It looks good, but made everything more complex. Currently, we have three types of inheritance 

* `Class.new(Transformer)` in `[]` because we don't need to inherit transproc
* `Class.new(self)` in `self` because we want to inherit both `container` and `transproc`
* `self[container]` in `create` because we don't want to inherit transproc

What's about prevent inheritance of transproc's for simplicity?

**Comment by Kukunin** [comment]
Also I've got some doubts about Transformer architecture. We actually don't need instance level there. We can even overwrite it with

```ruby
def self.new
  transproc
end
```

and will have nothing broken. If we don't need instance level - we can migrate everything from class level to instance level. And define `#build` level to return transproc.

**Comment by Kukunin** [comment]
Reverted transproc inheritance. Probably we can add a method for explicit inheritance if it will be needed

File: lib/transproc/transformer/class_interface.rb [hunk 1]
```diff
@@ -53,6 +53,33 @@ def container(container = ::Transproc::Undefined)
...
+      #
+      # @return [Function] Composed transproc
+      #
+      # @api public
+      def define(&block)
+        return transproc unless block_given?
+
+        Class.new(self).tap { |klass| klass.instance_eval(&block) }.transproc
+      end
+      alias build define
```
Comment (solnic) [inline lib/transproc/transformer/class_interface.rb]: What's the reason for the alias? `define` should be enough :)

Comment (Kukunin) [inline lib/transproc/transformer/class_interface.rb]: ROM Mapper has `#build` for initialization. I thought it doesn't hurt to add this. 
`#build` makes more sense without block while `#define` with block. 
In future we may refactor instance-level and `#new` will sounds weird

**Pull request merged**